### ISSUE TITLE
max_steps won't change when loading Imitation Learning model

### ISSUE THREAD

**Issue by JaAlSaDev** [Unity-Technologies/ml-agents#1863]
I trained my agents via Imitation Learning until the simulation steps reached `max_steps`, I had `max_steps`'s value set as 10,000.

I want to continue training, so I changed  `max_steps`'s value into 50,000 steps.
![DefaultHype1](https://user-images.githubusercontent.com/46474223/54905459-21585500-4ef2-11e9-8bc2-1c5d07fb116d.png)

I loaded the trained model:
![Train Command](https://user-images.githubusercontent.com/46474223/54905225-8f504c80-4ef1-11e9-9fcd-c0b825c5e286.PNG)

however, my change in `max_steps` was not detected, which won't allow me to continue the training.

![HyperparametersAnaconda](https://user-images.githubusercontent.com/46474223/54905308-c6266280-4ef1-11e9-97b1-52a5ad743834.PNG)

I'm using ML-Agents Beta 0.5.0a.

**Comment by awjuliani** [comment]
Hi @JaAlSaDev 

Changing the `max_steps` like this is not recommended. There are a number of internal variables which are based on the previous `max_step`, which when changed are no longer valid. I would recommend in general to train using a `max_step` as long or longer than you think you will need, and simply end the training session early when you receive the average reward you would like for your agent.

**Comment by JaAlSaDev** [comment]
Thanks for responding, @awjuliani!

I figured out what my problem was, the PushBlockIL scene has a brain called StudentBrain, I gave the same name to one of the brains in my scenes and added it to the configuration file `trainer_config.yaml`, So, the configuration file had two StudentBrains. I had to remove the configuration for PushBlockIL's StudentBrain.

![image](https://user-images.githubusercontent.com/46474223/54973528-54a5ed00-4fa1-11e9-9353-3371c4a997f8.png)


I'll close this issue, and I'll keep your recommendation in mind.
